File: 2025/zig/src/main.zig
const std = @import("std");
const _2025 = @import("_2025");

pub fn main() !void {
    var stdout_buffer: [1024]u8 = undefined;
    var stdout_writer = std.fs.File.stdout().writer(&stdout_buffer);
    const stdout = &stdout_writer.interface;
    const gpa = std.heap.page_allocator;

    for (_2025.days) |day| {
        const result = try _2025.runDayWithMetrics(gpa, day);

        if (!result.skipped) {
            try stdout.print("Day {s} part {s}: {s} (took {s})\n", .{ day.name, "1", result.part_one_result, result.part_one_time });
            try stdout.print("Day {s} part {s}: {s} (took {s})\n", .{ day.name, "2", result.part_two_result, result.part_two_time });
            try stdout.flush();
        }
    }
}

File: 2025/go/main.go
package main

import (
	"fmt"
	"io"
	"log"
	"os"
	"time"
)

func getInput(day string) string {
	data, err := os.ReadFile("../shared/" + day + ".txt")
	if err != nil {
		log.Fatal(err)
	}

	return string(data)
}

type Day struct {
	name    string
	PartOne func(string) string
	PartTwo func(string) string
}

func NotYetImplemented(_ string) string {
	return "not yet implemented"
}

func main() {
	days := [12]Day{
		{"1", DayOnePartOne, DayOnePartTwo},
		{"2", DayTwoPartOne, DayTwoPartTwo},
		{"3", DayThreePartOne, DayThreePartTwo},
		{"4", DayFourPartOne, DayFourPartTwo},
		{"5", DayFivePartOne, DayFivePartTwo},
		{"6", DaySixPartOne, DaySixPartTwo},
		{"7", DaySevenPartOne, DaySevenPartTwo},
		{"8", DayEightPartOne, DayEightPartTwo},
		{"9", DayNinePartOne, DayNinePartTwo},
		{"10", DayTenPartOne, DayTenPartTwo},
		{"11", DayElevenPartOne, DayElevenPartTwo},
		{"12", DayTwelvePartOne, DayTwelvePartTwo},
	}

	for _, day := range days {
		input := getInput(day.name)

		if len(input) == 0 {
			continue
		}

		before := time.Now()
		partOneResult := day.PartOne(input)
		between := time.Now()
		partTwoResult := day.PartTwo(input)
		after := time.Now()
		partOneDuration := between.Sub(before)
		partTwoDuration := after.Sub(between)

		line := fmt.Sprintf("Day %v part 1 (took %v): %v\n", day.name, partOneDuration, partOneResult)
		if _, err := io.WriteString(os.Stdout, line); err != nil {
			log.Fatal(err)
		}
		line = fmt.Sprintf("Day %v part 2 (took %v): %v\n", day.name, partTwoDuration, partTwoResult)
		if _, err := io.WriteString(os.Stdout, line); err != nil {
			log.Fatal(err)
		}
	}
}
